kubeflow/trainer/rhai/traininghub.py

```
@@ -186,6 +771,21 @@ def get_trainer_cr_from_training_hub_trainer(
 
         algorithm_name = trainer.algorithm.value
         raw_code = _render_algorithm_wrapper(algorithm_name, trainer.func_args)
+
+        # Inject progress tracking code if enabled
+        if trainer.enable_progression_tracking:
+            # Use the same default as the wrapper if ckpt_output_dir is not provided.
+            ckpt_dir = "/tmp/checkpoints"
```

Ruff (0.14.8)

Probable insecure usage of temporary file or directory: "/tmp/checkpoints"

(S108)